flang/test/Lower/OpenMP/wsloop-tile.f90

```diff
@@ -0,0 +1,30 @@
+! This test checks lowering of OpenMP DO Directive(Worksharing) with collapse.
+
+! RUN: bbc -fopenmp -fopenmp-version=51 -emit-hlfir %s -o - | FileCheck %s
+
+!CHECK-LABEL:   func.func @_QQmain() attributes {fir.bindc_name = "wsloop_tile"} {
+program wsloop_tile
+  integer :: i, j, k
+  integer :: a, b, c
+  integer :: x
+
+  a=30
+  b=20
+  c=50
+  x=0
+
+  !CHECK: omp.loop_nest
+  !CHECK-SAME: tiles(2, 5, 10)
```

Comment:
Can you add more details check lines for the loop nest structure?

Looking at the generated IR, we have 3 ops to represent the loop:
1. The `i` loop is an `omp.loop_nest` with a `tile` clause attached to it.
2. The `j` and `k` loops are `fir.do_loop` ops.

I kind of expected that the 3 loops will be collapsed into one `omp.loop_nest` op. But I am probably still missing the big picture here.

Reply:
It should be a single omp.loop_nest op, if not then it is a bug.

Reply:
I ran it on my machine and I'm getting this:
    omp.wsloop private(@_QFtripleloopEi_private_i32 %8#0 -> %arg1, @_QFtripleloopEj_private_i32 %10#0 -> %arg2, @_QFtripleloopEk_private_i32 %12#0 -> %arg3 : !fir.ref<i32>, !fir.ref<i32>, !fir.ref<i32>) {
      omp.loop_nest (%arg4, %arg5, %arg6) : i32 = (%c1_i32, %c1_i32_1, %c1_i32_3) to (%14, %15, %16) inclusive step (%c1_i32_0, %c1_i32_2, %c1_i32_4) tiles(2, 5, 10) {

Reply:
No problem, I added checks so that collapse(N) or tiles(1..N) makes sure that N is not greater than the number of loops in the omp.loop_nest op. It doesn't take into account the interaction between tiling and collapse, but we could add this also.